### PR TITLE
Create `Channel` from `ConsensusChannel` when direct defunding

### PR DESCRIPTION
**Pull request by geoknee**
Towards #598

### CODE REVIEW

File: client/engine/engine.go [hunk 1]
```diff
@@ -389,10 +391,13 @@ func (e *Engine) constructObjectiveFromMessage(id protocols.ObjectiveId, ss stat
 		}
 		return &vfo, nil
 	case directdefund.IsDirectDefundObjective(id):
-		ddfo, err := directdefund.ConstructObjectiveFromState(ss.State(), e.store.GetChannelById)
+		ddfo, err := directdefund.ConstructObjectiveFromState(ss.State(), e.store.GetConsensusChannelById)
 		if err != nil {
 			return &directdefund.Objective{}, fmt.Errorf("could not create direct defund objective from message: %w", err)
 		}
+
+		// TODO Destroy / Disable / Disown the associated ConsensusChannel
```
Comment (geoknee) [inline client/engine/engine.go:399]: This should become a new issue or be tracked in #598

File: protocols/directdefund/directdefund.go [hunk 2]
```diff
@@ -56,17 +56,25 @@ func isInConsensusOrFinalState(c *channel.Channel) (bool, error) {
 // GetChannelByIdFunction specifies a function that can be used to retreive channels from a store.
 type GetChannelByIdFunction func(id types.Destination) (channel *channel.Channel, ok bool)
 
+// todo: #420 assume name and godoc from GetTwoPartyLedgerFunction
```
Comment (geoknee) [inline protocols/directdefund/directdefund.go:59]: this should be done now, #420 is closed already

File: protocols/directdefund/directdefund.go [hunk 3]
```diff
@@ -56,17 +56,25 @@ func isInConsensusOrFinalState(c *channel.Channel) (bool, error) {
...
+	getConsensusChannel GetTwoPartyConsensusLedgerFunction,
 ) (Objective, error) {
-	c, ok := getChannel(channelId)
-
+	cc, ok := getConsensusChannel(channelId)
 	if !ok {
 		return Objective{}, fmt.Errorf("could not find channel %s", channelId)
 	}
+
+	c, err := CreateChannelFromConsensusChannel(*cc)
```
Comment (geoknee) [inline protocols/directdefund/directdefund.go:73]: This is the nub of the change.

**Review by lalexgap — APPROVED** [review]

File: client/engine/engine.go (hunk 1 above)
Comment (lalexgap) [reply client/engine/engine.go:399]: Created #607

File: protocols/directdefund/directdefund.go (hunk 2 above)
Comment (lalexgap) [reply protocols/directdefund/directdefund.go]: Done in https://github.com/statechannels/go-nitro/pull/602/commits/19b0cf43f2778592a0b223b7e48732876d9d6c25

**Pull request merged**